linalg/hypre.cpp

```diff
@@ -5269,14 +5269,60 @@ HypreAMS::HypreAMS(const HypreParMatrix &A, HypreParMatrix *G_,
    MFEM_ASSERT(y != NULL, "");
    int sdim = (z == NULL) ? 2 : 3;
    int cycle_type = 13;
-
    MakeSolver(sdim, cycle_type);
 
    HYPRE_ParVector pz = z ? static_cast<HYPRE_ParVector>(*z) : NULL;
    HYPRE_AMSSetCoordinateVectors(ams, *x, *y, pz);
    HYPRE_AMSSetDiscreteGradient(ams, *G);
 }
 
+void HypreAMS::Init(ParFiniteElementSpace *edge_fespace)
+{
+   ParMesh *pmesh = edge_fespace->GetParMesh();
+   int dim = pmesh->Dimension();
```

Comment:
These could be `const int`.

Reply:
Given the lack of `const` and `constexpr` used in the other `HypreAMS` member functions I'm inclined to leave it as is. For consistency with the conventions throughout `hypre.cpp`  for the other `HypreSolver` classes, and in the interest of not introducing a lot of unrelated changes, I think it is fine to keep non-`const` for now. If other reviewers think otherwise I can go ahead and make the change throughout.